.github/workflows/ci.yml

```diff
@@ -0,0 +1,86 @@
+name: CI
+
+on:
+  push:
+    branches:
+    - main
+    - stable
+    - v*.*.*
+    tags:
+    - v*
+  pull_request:
+  workflow_dispatch:
+  schedule:
+  - cron: 37 8 * * 1
```

Comment:
It'll be helpful to run this every week, since that can help us narrow down when new test or doc build failures show up.